core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -3436,6 +3437,20 @@ object SparkContext extends Logging {
     supplement(DRIVER_JAVA_OPTIONS)
     supplement(EXECUTOR_JAVA_OPTIONS)
   }
+
+  private def supplementBlasOptions(conf: SparkConf): Unit = {
+    conf.getOption("spark.ml.allowNativeBlas").foreach { allowNativeBlas =>
```

Comment:
I am not sure whether we can use a env variable like `MKL_NUM_THREADS=1` `OPENBLAS_NUM_THREADS=1`

Reply:
I skimmed the codebase https://github.com/luhenry/netlib and found neither sys prop nor env to disable native blas loading, so I need to introduce a new one. do you mean env var is preferred over sys prop?

Reply:
I guess I mean, does this have to propagate via sys property at all versus just being used as a config in the code directly? But probably it is hard to plumb through access to the config object

Reply:
@srowen I replied in https://github.com/apache/spark/pull/49986#discussion_r1958982565